lms/djangoapps/courseware/grades.py

```diff
@@ -126,9 +127,18 @@ def grade(student, request, course, keep_raw_scores=False):
     """
     Wraps "_grade" with the manual_transaction context manager just in case
     there are unanticipated errors.
+    Send a signal to update the minimum grade requirement status.
     """
     with manual_transaction():
-        return _grade(student, request, course, keep_raw_scores)
+        grade_summary = _grade(student, request, course, keep_raw_scores)
+        MIN_GRADE_REQUIREMENT_STATUS.send_robust(
+            sender=None,
+            request=request,
```

Comment:
I got 19 failed test cases. That's why I passed the request object to signal and from there picked the user name.